File: src/main/java/net/ndrei/teslapoweredthingies/machines/fluidburner/FluidBurnerFuelRecipe.java
package net.ndrei.teslapoweredthingies.machines.fluidburner;

import net.minecraftforge.fluids.Fluid;

/**
 * Created by CF on 2017-01-09.
 */
public class FluidBurnerFuelRecipe {
    public final Fluid fluid;
    public final int amount;
    public final int baseTicks;

    public FluidBurnerFuelRecipe(Fluid fluid, int amount, int baseTicks) {
        this.fluid = fluid;
        this.amount = amount;
        this.baseTicks = baseTicks;
    }
}

File: src/main/kotlin/net/ndrei/teslapoweredthingies/machines/fluidburner/FluidBurnerFuelRecipe.kt
package net.ndrei.teslapoweredthingies.machines.fluidburner

import net.minecraft.util.ResourceLocation
import net.minecraftforge.fluids.Fluid
import net.ndrei.teslapoweredthingies.MOD_ID
import net.ndrei.teslapoweredthingies.common.BaseTeslaRegistryEntry

/**
 * Created by CF on 2017-06-30.
 */
class FluidBurnerFuelRecipe(val fluid: Fluid, val amount: Int, val baseTicks: Int)
    : BaseTeslaRegistryEntry<FluidBurnerFuelRecipe>(FluidBurnerFuelRecipe::class.java, ResourceLocation(MOD_ID, "fluid_${fluid.name}"))
